intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/highlighting/SemanticHighlightVisitor.xtend

```diff
@@ -73,8 +70,9 @@ class SemanticHighlightVisitor implements HighlightVisitor {
 	}
 	
 	override clone() {
-		cloneProvider.get => [
+		getClass.newInstance => [
```

Comment:
Implementing Cloneable and calling super.clone should make sure that all fields are copied regardless of specializations in subtypes. I would expect the clone doesn't have a languageID od highlightCalculator right now